Casks/t/telegram.rb

```diff
@@ -9,9 +9,14 @@
 
   livecheck do
     url "https://osx.telegram.org/updates/versions.xml"
-    strategy :page_match do |page|
-      page.scan(/Telegram[._-](\d+(?:\.\d+)+)\.(\d{6})\.app\.zip/)
-          .map { |matches| "#{matches[0]},#{matches[1]}" }
+    regex(/Telegram[._-](\d+(?:\.\d+)+)\.(\d{6})\.app\.zip/)
```

Comment:
```suggestion
    regex(/Telegram[._-]v?(\d+(?:\.\d+)+)\.(\d{4,})\.app\.zip/)
```

This will fail to match if/when the last number exceeds six digits, so there's something to be said for allowing for a bit more flexibility here. It's unlikely that the digits in the first part of the version will exceed four, so we could use `\d{4,}` here (or `\d{6,}` at the very least, as it will continue to match if the number becomes longer).